desc/magnetic_fields/_core.py

```diff
@@ -1152,6 +1477,15 @@ def _atleast_4d(x):
         self._derivs["BR"] = self._approx_derivs(self._BR)
         self._derivs["Bphi"] = self._approx_derivs(self._Bphi)
         self._derivs["BZ"] = self._approx_derivs(self._BZ)
+        if AR is not None and Aphi is not None and AZ is not None:
+            AR, Aphi, AZ = map(_atleast_4d, (AR, Aphi, AZ))
+            assert AR.shape == Aphi.shape == AZ.shape == shape
+            self._AR = AR
```

Comment:
if they only get assigned some of the time but they're listed under io_attrs, I tihnk that will cause a bunch of warnings every time you load/save a spline field without them. Probably better to assign None in the case.